assignment-client/src/assets/UploadAssetTask.cpp

```diff
@@ -0,0 +1,99 @@
+//
+//  UploadAssetTask.cpp
+//  assignment-client/src/assets
+//
+//  Created by Stephen Birarda on 2015-08-28.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "UploadAssetTask.h"
+
+#include <QtCore/QBuffer>
+#include <QtCore/QFile>
+
+#include <AssetUtils.h>
+#include <NodeList.h>
+#include <NLPacketList.h>
+
+
+UploadAssetTask::UploadAssetTask(QSharedPointer<NLPacketList> packetList, SharedNodePointer senderNode,
```

Comment:
The SendAssetTask doesn't handle the packet parsing. I don't have a strong opinion on where that should happen, but we should probably make it the same for both.